libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java

```diff
@@ -471,6 +475,15 @@ public void onNewBannerInstructions(@NotNull BannerInstructions bannerInstructio
     }
   };
 
+  private TripSessionStateObserver tripSessionStateObserver = new TripSessionStateObserver() {
+    @Override
+    public void onSessionStateChanged(@NotNull TripSessionState tripSessionState) {
+      if (tripSessionState == TripSessionState.STOPPED) {
```

Comment:
Our observers are implemented to be reactive which means that as soon as you register it current state is sent 👀 https://github.com/mapbox/mapbox-navigation-android/blob/a9369765cc47964bc05fe2e918d72ad22b4feae6/libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt#L272 That means that `onSessionStateChanged` will be fired initially with `TripSessionState.STOPPED`. Could that be a problem in this particular scenario / originate unexpected behavior if `navigationViewEventDispatcher.onNavigationFinished();` is called right away the `TripSessionStateObserver` is registered?

Reply:
that's a very good question.

the good thing is we are good here. We initialize the viewmodel before `navigationViewEventDispatcher.initializeListeners` to setup the `navigationListener`. So when the **TripSessionState.STOPPED** fired initially, the navigationListener is **NULL** still.